core/blockchain.go

```diff
@@ -750,14 +751,31 @@ func (bc *BlockChain) ExportN(w io.Writer, first uint64, last uint64) error {
 	log.Info("Exporting batch of blocks", "count", last-first+1)
 
 	start, reported := time.Now(), time.Now()
+	var stReceipts []*types.ReceiptForStorage
+
 	for nr := first; nr <= last; nr++ {
 		block := bc.GetBlockByNumber(nr)
 		if block == nil {
 			return fmt.Errorf("export failed on #%d: not found", nr)
 		}
+
 		if err := block.EncodeRLP(w); err != nil {
 			return err
 		}
+		if receipts {
+			receipts := bc.GetReceiptsByHash(block.Hash())
+			if receipts == nil {
+				return fmt.Errorf("nil receipts for block %d %s\n", nr, block.Hash())
+			}
+			stReceipts = stReceipts[:0]
```

Comment:
```suggestion
var stReceipts []*types.ReceiptForStorage
```
or maybe: 
```
 stReceipts := make([]*types.ReceiptForStorage, 0, len(receipts))
 ```

Reply:
cool, done (I wanted to avoid the reallocs out of habit, but admittedly pointless as this isn't GC-bound)